include/tunnler/room_member.h

```diff
@@ -23,6 +24,14 @@
 // It also has to be used if you host a game yourself (You'd create both, a Room and a RoomMembership for yourself)
 class RoomMember final {
 public:
+    using Callback = std::shared_ptr<std::function<void()> >;
```

Comment:
for API look ups i use cppreference. But that isn't from there. The solution for that is adopted from [stackoverflow](http://stackoverflow.com/questions/23997287/how-to-handle-thread-safe-callback-registration-and-execution-in-c)